vm/src/stdlib/os.rs

```diff
@@ -793,6 +793,21 @@ mod _os {
             )
         }
 
+        #[pymethod]
+        fn inode(&self, vm: &VirtualMachine) -> PyResult<u64> {
+            stat_inner(
+                PathOrFd::Path(PyPathLike {
+                    path: self.entry.path(),
+                    mode: OutputMode::String,
+                }),
+                DirFd::default(),
+                FollowSymlinks(false),
+            )
+            .map_err(|e| e.into_pyexception(vm))?
+            .ok_or_else(|| vm.new_value_error("embedded null character".to_owned()))
+            .map(|stat| stat.st_ino)
```

Comment:
I think on unix, you can use [`std::os::unix::fs::DirEntryExt::ino`](https://doc.rust-lang.org/stable/std/os/unix/fs/trait.DirEntryExt.html#tymethod.ino), and on windows, you can cache the result of `stat_inner` with an `AtomicCell<Option<u64>>` on DirEntry

Reply:
The new version uses [`std::os::unix::fs::DirEntryExt::ino`](https://doc.rust-lang.org/stable/std/os/unix/fs/trait.DirEntryExt.html#tymethod.ino) for unix and use `stat_inner` for other platform. Thanks for the nice suggestion.